main.tf

```diff
@@ -57,20 +74,40 @@ resource "tfe_team_access" "default" {
 }
 
 resource "tfe_variable" "aws_access_key_id" {
+  count        = var.workspace_authentication_method == "user" ? 1 : 0
   key          = "AWS_ACCESS_KEY_ID"
-  value        = module.workspace_account.access_key_id
+  value        = module.workspace_account[count.index].access_key_id
   category     = "env"
   workspace_id = tfe_workspace.default.id
 }
 
 resource "tfe_variable" "aws_secret_access_key" {
+  count        = var.workspace_authentication_method == "user" ? 1 : 0
   key          = "AWS_SECRET_ACCESS_KEY"
-  value        = module.workspace_account.secret_access_key
+  value        = module.workspace_account[count.index].secret_access_key
   category     = "env"
   sensitive    = true
   workspace_id = tfe_workspace.default.id
 }
 
+resource "tfe_variable" "aws_role" {
+  count        = var.workspace_authentication_method == "role" ? 1 : 0
+  key          = "iam_role"
+  value        = module.workspace_account_role[count.index].arn
+  category     = "terraform"
+  sensitive    = true
+  workspace_id = tfe_workspace.default.id
+}
+
+resource "tfe_variable" "external_id" {
+  count        = var.workspace_authentication_method == "role" ? 1 : 0
+  key          = "external_id"
+  value        = local.tfc_agent_externalid
```

Comment:
This should be replaced with the generated alphanumeric string.

Reply:
It already is an autogenerated integer, but it happens in the AVM module so it can keep the state. Has been discussed with Marwin.